### PR TITLE
Add CRUD user and api that check used email,username

### PR DESCRIPTION
**Pull request by gimochi**
## Related Issues
#98, #99, #100, #102 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (one that would cause existing functionality to change)

## Testing
- [ ] Tested in some way, if yes, the description follows.

### CODE REVIEW

File: src/controllers/User/UserCheckUsedController.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  if (userSelected.length == 0) {
+    //TODO return value 는 정해져야함
+    return "NotUsedUsername";
+  } else {
+    return "UsedUsername";
+  }
+}
+
+export const checkEmailUsed = async (req) => {
+  const userSelected = await UserSelectDAO.selectUserByEmail(req.email)
```
Comment (enginehenryed) [inline src/controllers/User/UserCheckUsedController.ts]: ;

File: src/controllers/User/UserCheckUsedController.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import db from '../../database';
+import * as UserInsertDAO from '../../dao/User/UserInsertDAO';
+import * as UserSelectDAO from '../../dao/User/UserSelectDAO';
+import { transaction } from '../../database/databaseUtils';
+import { authConfig } from '../../config/marmoym-config';
+import MarmoymError from "../../models/MarmoymError";
+import ErrorType from '../../models/ErrorType';
+
+export const checkUsernameUsed = async (req) => {
+  const userSelected = await UserSelectDAO.selectUserByUsername(req.username)
```
Comment (enginehenryed) [inline src/controllers/User/UserCheckUsedController.ts]: ;

File: src/routes/URL.ts [hunk 3]
```diff
@@ -1,5 +1,8 @@
 export const DEFINITION_ROUTE = `/definitions`;
```
Comment (enginehenryed) [inline src/routes/URL.ts]: `ROUTE` is so obvious. We'd better get rid of all of them.

File: src/services/errorHandleService.ts [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import * as winston from 'winston';
 
 export const handleError = (err, req, res, next) => {
+  winston.debug(err);
+  
   if (err.constructor.name == 'MarmoymError') {
```
Comment (enginehenryed) [inline src/services/errorHandleService.ts]: `===`

File: src/routes/RequestTypes.ts [hunk 5]
```diff
@@ -52,6 +52,9 @@ export namespace DefinitionRequest {
 }
```
Comment (enginehenryed) [inline src/routes/RequestTypes.ts]: Please leave the json data you tested as comment

**Review by eldenpark — CHANGES_REQUESTED** [review]
You're adjusting fast. Soon you may not need the feedbacks

**Pull request merged**